control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -313,12 +313,14 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 
 	broker.Status.Address.URL = nil
 
+	ingressHost := network.GetServiceHostname(r.Env.IngressName, r.Reconciler.DataPlaneNamespace)
+
 	//  Rationale: after deleting a topic closing a producer ends up blocking and requesting metadata for max.block.ms
 	//  because topic metadata aren't available anymore.
 	// 	See (under discussions KIPs, unlikely to be accepted as they are):
 	// 	- https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=181306446
 	// 	- https://cwiki.apache.org/confluence/display/KAFKA/KIP-286%3A+producer.send%28%29+should+not+block+on+metadata+update
-	address := receiver.Address(r.IngressHost, broker)
+	address := receiver.Address(ingressHost, broker)
```

Comment:
> Recompute broker IngressHost every time - needed for future dynamic dataplanes; instead of computing it once and setting it as a struct field.

As I wrote in the PR desc, I will create the reconciler instance dynamically with each reconcilation request. This was necessary because of testing and some other reason (that I don't remember right now).

Also, this very cheap call.

> so ingressHost == r.IngressHost

Yeah, "ingressHost == r.IngressHost", otherwise I wouldn't make this change.
This is a simple refactoring that doesn't change any business logic.

Reply:
With this change, ingressHost is recalculated in regular KafkaBroker with every reconciliation request. ...which will be the same with the future broker class implementation.

Reply:
> As I wrote in the PR desc, I will create the reconciler instance dynamically with each reconcilation request. This was necessary because of testing and some other reason (that I don't remember right now).

> Recompute broker IngressHost every time - needed for future dynamic dataplanes; instead of computing it once and setting it as a struct field.
Computing: string building for the ingress service name
Prefactoring for KafkaBroker with dataplane-per-namespace

This doesn't tell me the reason why we need to recompute it, so I still don't get it, sorry

Reply:
/close

Closing as I see you're very strong about rejecting this change.

Reply:
Ok, your comment does help l to understand the context here, sorry I didn't really get what was the reason, I see the value of decoupling the 2 but if it's not really needed, let's get rid of those variables.

/reopen